lib/modem_info/modem_info_json.c

```diff
@@ -97,67 +57,192 @@ static int json_add_str(cJSON *parent, const char *str, const char *item)
 	return json_add_obj(parent, str, json_str);
 }
 
-int modem_info_json_string_get(char *buf)
+static int json_add_data(struct lte_param *param, cJSON *json_obj)
 {
-	u16_t param_value;
-	size_t len = 0;
-	size_t total_len = 0;
-	int ret = 0;
-	char data_buffer[MODEM_INFO_MAX_RESPONSE_SIZE] = {0};
 	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE];
-	cJSON *data_obj = cJSON_CreateObject();
 	enum at_param_type data_type;
+	int total_len = 0;
+	int ret;
+
+	memset(data_name, 0, MODEM_INFO_MAX_RESPONSE_SIZE);
+	ret = modem_info_name_get(param->type,
+				data_name);
+	if (ret < 0) {
+		LOG_DBG("Data name not obtained: %d\n", ret);
+		return -EINVAL;
+	}
+
+	data_type = modem_info_type_get(param->type);
 
-	if (data_obj == NULL) {
+	if (data_type < 0) {
+		return -EINVAL;
+	}
+
+	if (data_type == AT_PARAM_TYPE_STRING &&
+	    param->type != MODEM_INFO_AREA_CODE) {
+		total_len += strlen(param->string);
+		ret += json_add_str(json_obj, data_name, param->string);
+	} else {
+		total_len += sizeof(u16_t);
+		ret += json_add_num(json_obj, data_name, param->value);
+	}
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	return total_len;
+}
+
+static int network_data_add(struct network_param *network, cJSON *json_obj)
+{
+	char data_name[MODEM_INFO_MAX_RESPONSE_SIZE];
+	size_t total_len;
+	int ret;
+	int len;
+
+	if (network == NULL || json_obj == NULL) {
+		return -EINVAL;
+	}
+
+	total_len = json_add_data(&network->cur_band, json_obj);
+	total_len += json_add_data(&network->sup_band, json_obj);
+	total_len += json_add_data(&network->area_code, json_obj);
+	total_len += json_add_data(&network->mcc, json_obj);
+	total_len += json_add_data(&network->mnc, json_obj);
+	total_len += json_add_data(&network->ip_address, json_obj);
+	total_len += json_add_data(&network->ue_mode, json_obj);
+
+	len = modem_info_name_get(network->cellid_hex.type, data_name);
+	data_name[len] =  '\0';
+	ret = json_add_num(json_obj, data_name, network->cellid_dec);
+
+	if (ret) {
+		LOG_DBG("Unable to add the cell ID.\n");
+	} else {
+		total_len += sizeof(double);
+	}
+
+	len = modem_info_name_get(network->lte_mode.type, data_name);
+	data_name[len] =  '\0';
+	ret = json_add_bool(json_obj, data_name, network->lte_mode.value);
+
+	if (ret) {
+		LOG_DBG("Unable to add the LTE-M system mode.\n");
+		return ret;
+	} else {
+		total_len += sizeof(bool);
+	}
+
+	len = modem_info_name_get(network->nbiot_mode.type, data_name);
+	data_name[len] =  '\0';
+	ret = json_add_bool(json_obj, data_name, network->nbiot_mode.value);
+
+	if (ret) {
+		LOG_DBG("Unable to add the NB-IoT system mode.\n");
+		return ret;
+	} else {
+		total_len += sizeof(bool);
+	}
+
+	len = modem_info_name_get(network->gps_mode.type, data_name);
+	data_name[len] =  '\0';
+	ret = json_add_bool(json_obj, data_name, network->gps_mode.value);
+
+	if (ret) {
+		LOG_DBG("Unable to add the GPS system mode.\n");
```

Comment:
`\n` can preferably be dropped in log messages, it's added by the log module

Reply:
Good to know. I've removed `\n` for all `LOG_DBG` instances in the modem informer.